android-lib/src/androidTest/java/com/kinvey/androidTest/ClientBuilderTest.java

```diff
@@ -0,0 +1,164 @@
+import com.kinvey.android.Client;
+import com.kinvey.android.model.User;
+import com.kinvey.androidTest.model.TestUser;
+import com.kinvey.java.KinveyException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Created by yuliya on 08/02/17.
+ */
+
+@RunWith(AndroidJUnit4.class)
+public class ClientBuilderTest {
+
+    private static final String BASE_URL_DEFAULT = "https://baas.kinvey.com/";
+    private static final String TEST_APP_KEY = "app_key";
+    private static final String TEST_APP_SECRET = "app_secret";
+    private static final String KINVEY_PROPERTIES = "kinvey.properties";
+
+    private Client client;
+    private Context mContext;
+
+    @Before
+    public void setUp() throws InterruptedException, IOException {
+        mContext = new RenamingDelegatingContext(InstrumentationRegistry.getInstrumentation().getTargetContext(), "test_");
+    }
+
+    @Test
+    public void testSetDeltaSetCache() throws IOException {
+        assertTrue(new Client.Builder(mContext).setDeltaSetCache(true) instanceof Client.Builder);
+    }
+
+    @Test
+    public void testSetDeltaSetCacheTrue() throws IOException {
+        client = new Client.Builder(mContext).setDeltaSetCache(true).build();
+        assertTrue(client.isUseDeltaCache());
+    }
+
+    @Test
+    public void testSetDeltaSetCacheFalse() throws IOException {
+        client = new Client.Builder(mContext).setDeltaSetCache(false).build();
+        assertFalse(client.isUseDeltaCache());
+    }
+
+    @Test
+    public void testSetDeltaSetCacheDefault() throws IOException {
+        client = new Client.Builder(mContext).build();
+        assertFalse(client.isUseDeltaCache());
+    }
+
+    @Test
+    public void testSetUserClass() throws IOException {
+        client = new Client.Builder(mContext).setUserClass(TestUser.class).build();
+        assertTrue(client.getUserClass() == TestUser.class);
+    }
+
+    @Test
+    public void testSetUserClassDefault() throws IOException {
+        client = new Client.Builder(mContext).build();
+        assertTrue(client.getUserClass() == User.class);
+    }
+
+    @Test
+    public void testSetBaseUrl() throws IOException {
+        String url = "https://base_url/";
+        client = new Client.Builder(mContext).setBaseUrl(url).build();
+        assertTrue(client.getBaseUrl().equals(url));
+    }
+
+    @Test
+    public void testSetWrongBaseUrl() throws IOException {
+        String url = "base_url";
+        try {
+            client = new Client.Builder(mContext).setBaseUrl(url).build();
+            assertTrue(false);
+        } catch (KinveyException e) {
+            assertNotNull(e);
```

Comment:
@yuliya-guseva it'll be good for this assert to be a little more specific e.g. what is the `KinveyException`, is there a particular code or debug message expected etc.